feat(apple): How to disable collecting IP address (#4786)
getsentry/sentry-docs: src/platforms/common/enriching-events/identify-user.mdx

The user's IP address. If the user is unauthenticated, Sentry uses the IP address as a unique identifier for the user. Sentry will attempt to pull this from the HTTP request data (`request.env.REMOTE_ADDR` field in JSON), if available. That might require <PlatformIdentifier name="send-default-pii" /> set to `true` in the SDK options.

> If the user's `ip_address` is set to `"{{auto}}"`, Sentry will infer the IP address from the connection between your app and Sentry's server. <PlatformSection supported={["apple"]}>Due to backward compatibility concerns, Sentry sets the IP address to `"{{auto}}"` out of the box for Apple. Therefore setting <PlatformIdentifier name="send-default-pii" /> set to `false` won't stop Sentry from collecting users' IP addresses. Read the paragraphs below to find out how to opt out of collecting users' IP addresses.</PlatformSection>

Reviewer: ```suggestion
If the user's `ip_address` is set to `"{{auto}}"`, Sentry will infer the IP address from the connection between your app and Sentry's server. <PlatformSection supported={["apple"]}>Due to backward compatibility concerns, Sentry sets the IP address to `"{{auto}}"` out of the box for Apple. Therefore setting <PlatformIdentifier name="send-default-pii" /> to `false` won't stop Sentry from collecting users' IP addresses. Read the paragraphs below to find out how to opt out of collecting users' IP addresses.</PlatformSection>
```

> If the user's `ip_address` is set to `"{{auto}}"`, Sentry will infer the IP address from the connection between your app and Sentry's server. <PlatformSection supported={["apple"]}>Due to backward compatibility concerns, Sentry sets the IP address to `"{{auto}}"` out of the box for Apple. Therefore setting <PlatformIdentifier name="send-default-pii" /> set to `false` won't stop Sentry from collecting users' IP addresses. Read the paragraphs below to find out how to opt out of collecting users' IP addresses.</PlatformSection>

Reviewer 2: ```suggestion
If the user's `ip_address` is set to `"{{auto}}"`, Sentry will infer the IP address from the connection between your app and Sentry's server. <PlatformSection supported={["apple"]}>Due to backward compatibility concerns, Sentry sets the IP address to `"{{auto}}"` out of the box for Apple. Therefore setting <PlatformIdentifier name="send-default-pii" /> to `false` won't stop Sentry from using the client connection IP addresses. Read the paragraphs below to find out how to opt out of using the client connection IP addresses.</PlatformSection>
```

If the field is omitted, the default value is `null`. However, due to backwards compatibility concerns, certain platforms (in particular JavaScript) have a different default value for `"{{auto}}"`. SDKs and other clients should not rely on this behavior and should set IP addresses or `"{{auto}}"` explicitly.

> To opt out of collecting users' IP addresses, you can go to your project settings, click on Security and Privacy, and enable Prevent Storing of IP Addresses or use Sentry's [server-side data](/product/data-management-settings/scrubbing/) scrubbing to remove `$user.ip_address`. Adding such a rule ultimately overrules any other logic.

Reviewer: ```suggestion
To opt out of collecting users' IP addresses, you can go to your project settings, click on "Security & Privacy", and enable "Prevent Storing of IP Addresses" or use Sentry's [server-side data](/product/data-management-settings/scrubbing/) scrubbing to remove `$user.ip_address`. Adding such a rule ultimately overrules any other logic.
```
